### DISCUSSION TITLE
Button to trigger get request

### DISCUSSION THREAD

**Discussion by 0xPiPo** [HomeSpan/HomeSpan#353] [Q&A] [answered]
I want to try to run a get request when I touch a button. How can that be done?

**Comment by HomeSpan** [comment]
I'm not quite sure I understand what you are trying to do, but if the idea is to communicate the status of a HomeSpan device back to your Home App via a button press, this is covered in Tutorial Example 15.

**Reply by 0xPiPo** [reply]
I actually have an ESP8266 as an IR server and I wanted my ESP32 with HomeSpan to have lets say a Button in the home app. When I press that button then I want it to trigger a GET request for example "192.168.X.X/sendircommand=15". Is it possible?

**Comment by HomeSpan** [comment, marked as answer]
Depending on exactly what you would like to control there may be appropriate Home App Services to use, such as a Light Service, Fan Service, etc. that will visually match your device.  Regardless of which you choose, when the buttons on the Home App are pressed, this will send an update command to HomeSpan on the ESP32.  You can then program the ESP32 to do just about anything you'd like.  You can, for example, have it control an IR LED - I use this a lot to control non-HomeKit compatible TVs.

You can also write code to send web GET commands to another client, such as your IR Server.  I have some examples of adding a web _server_ to HomeSpan (see the FAQ's section) but have not done anything with outbound requests.  However, producing a GET command should be very similar.   Note HomeSpan does not have anything built-in to do this, so you'll have to write your own web calls.  My recommendation would be to port your IR Sensor server to the ESP32 and then overlay HomeSpan to create a single HomeKit-compatible device.

**Reply by 0xPiPo** [reply]
Thanks a lot for the answer
